_gsocblogs/2022/blog_CompilerResearch-ManishKausik.md

```diff
@@ -22,6 +22,25 @@ As the Community Bonding period started, I got into fixing a few bugs in the [Cl
+
+## Coding Period 2
+Initially my second Pull Request integrated Enzyme within Clad as a shared object, but upon some discussion with my mentors, we decided to statically link Enzyme with Clad. My mentor helped me with this. Statically linking did bring up some issues in scheduling the Enzyme Passes correctly, which took some time for me to figure out. Thus the second PR did take some time for me to achieve, but when it was successfully merged, it was a milestone in the project as it gave a platform to call Enzyme within Clad.
+
+Now the stage was set to add support for differentiating different function prototypes with Enzyme in Clad. This was achieved by my third Pull Request [Add Support for Differentiating functions with both pointer/array type and primitive type parameters with Enzyme](https://github.com/vgvassilev/clad/pull/486). This PR completes initial Enzyme Reverse Mode integration with Clad. 
+
+Now that I had integrated Enzyme within Clad, to comprehensively complete my Project, I contributed to a few other Pull Requests which are mentioned here:
+
++ [Benchmark Clad and Enzyme](https://github.com/vgvassilev/clad/pull/491): This PR compares the execution time of Clad generated derivatives with Enzyme Generated derivatives with Google's Benchmark library.
++ [Treat Request to use Enzyme for Differentiating a Function to be different from a Clad differentiation](https://github.com/vgvassilev/clad/pull/492): This PR was a bug fix.
++ [Add Documentation for Integration of Enzyme with Clad](https://github.com/vgvassilev/clad/pull/494): This PR adds a page in the Clad Documentation explaining the details of the Project as well as the calling convention to use the features added by the project.
++ [Add tests to verify enzyme results with clad](https://github.com/vgvassilev/clad/pull/495): This PR verifies if the Gradients generated by Enzyme and Clad match to a certain fixed level of accuracy.
+
+## My Experience with GSoC
+This project gave me a good insight into Software Engineering practices used by the Open Source Community and a great experience in learning some State of the Art Compiler Libraries like Clang and LLVM. It was also a great opportunity to meet members of both Clad and Enzyme communities. Honestly, not everything about the Project went smoothly, as there were some unexpected delays in code reviews and I did face several coding challenges during the project. But thanks to timely help from my mentors and other members of the community I was able to achieve most deliverables that I had proposed in my GSoC Proposal. Overall, it was a great learning experience!
```

Comment:
When you say 'achieve most deliverables that I had proposed' that begs the question what are the things left and maybe outline them in future work.